### ISSUE TITLE
review accessibility of linked image tile

### ISSUE THREAD

**Issue by ntno** [ntno/mkdocs-terminal#107]
see https://webaim.org/blog/alt-text-and-linked-images/
comment from article:
> Title is intended to provide supplementary information. If title is necessary for accessibility, then it is therefor NOT supplementary. 

**and**

see https://usability.yale.edu/web-accessibility/articles/images
> Image links should describe the purpose of the link, and must never describe the image.

**and**

https://usability.yale.edu/web-accessibility/articles/links
> Image Links
> In general, content editors should avoid using images as links. If an image functions as a link, the image must have alt text that conveys the location and purpose of the link. The alt text should not describe the image. Treat image links as links, not as images.

also:
- https://www.w3.org/TR/WCAG20-TECHS/G196
- consider what should happen if a title is provided for both the link and the image

**Comment by ntno** [comment]
oof
https://www.powermapper.com/tests/screen-readers/labelling/a-img-null-alt/
if you have a link and an image inside the link but the image has a null alt tag it causes problems in 135 screen reader / browser combinations
`<a href='4567.html'><img src='1234.png' alt=''></a>`

**Comment by ntno** [comment]
https://www.powermapper.com/tests/screen-readers/labelling/a-img-alt/
when image is inside a link, the alt text should declare that it represents a link:

`<a href='4567.html'><img src='1234.png' alt='this is a link alt'></a>`

example ouput from screenreader:

"link graphic, this is a link alt"
or
"This is a link alt, graphic link"

**Comment by ntno** [comment]
https://www.w3.org/TR/WCAG20-TECHS/H30.html
W3 example 


```
<a href="routes.html">
   <img src="topo.gif" alt="Current routes at Boulders Climbing Gym" /> 
</a> 
```


if the link contains text then the alt should be null:
```
<a href="routes.html">
  <img src="topo.gif" alt="" />
  Current routes at Boulders Climbing Gym
</a>
```

**Comment by ntno** [comment]
some examples have the link text go before the image tag:

```
<a href="WMFP.pdf">
Woodend Music Festival Program
<img src="pdficon.gif" alt="PDF format"/>
</a>
```

**Comment by ntno** [comment]
consider adding longdesc attribute? https://www.w3.org/TR/WCAG20-TECHS/H45.html
this is a link to a location with additional description, does not contain the description itself

```
<img longdesc="thispage.html#desc" alt="Line graph of the number of subscribers" src="http://www.company/images/graph.png">
<div id="desc">
<h3>Long Description: Line graph of the number of subscribers</h3>
<!-- Full Description of Graph -->
<p>Long description ends.</p>
<div>
```

or 

```<p><img src="chart.gif" alt="a complex chart" longdesc="chartdesc.html"/></p>```

**Comment by ntno** [comment]
wait nvm longdesc is deprecated - https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/longDesc

**Comment by ntno** [comment]
maybe https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-details
?

**Comment by ntno** [comment]
maybe remove img_alt and replace with alt_text
then add an img_desc
img_desc would be for either: 
- descriptions too long for typical alt text (>100 char)
- description for an image when the image is being used as a link

**Comment by ntno** [comment]
issue^: no way to enforce that linked image's alt text indicates that it is a link
maybe just need to rely on documentation to guide this

**Comment by ntno** [comment]
how do **img_desc** and **caption** interact with each other
where should **img_desc** be displayed?

**Comment by ntno** [comment]
idea:
maybe a link underneath the caption
"Detailed Description"?  "Detailed Figure Description"?  
then **img_desc** would be a link to a location with additional information

problem:  in a tile grid we would have a bunch of links with the exact same name

**Comment by ntno** [comment]
footnotes could be useful here because they provide a back link but i don't want to force the use of footnotes
would be handy if it also worked for footnotes though?  test this

**Comment by ntno** [comment]
leaning towards appending the img_desc link to caption
both inside `<figcaption>`
